vmdb/spec/requests/api/tags_spec.rb

```diff
@@ -40,331 +42,313 @@ def app
+           "name"         => tag.classification.name,
+           "description"  => tag.classification.description,
+           "display_name" => "#{tag.category.description}: #{tag.classification.description}",
+           "category"     => {"name" => tag.category.name, "description" => tag.category.description}
+         }
+       )
+      end
     end
 
-    it "query all tags with categorization" do
-      basic_authorize @cfme[:user], @cfme[:password]
+    context "query all tags with categorization" do
+      before do
+        api_basic_authorize
 
-      @success = run_get "#{@cfme[:tags_url]}?expand=resources&attributes=categorization"
+        run_get "#{tags_url}?expand=resources&attributes=categorization"
+      end
 
-      expect(@success).to be_true
-      expect(@code).to eq(200)
-      results = @result["resources"]
-      expect(results.size).to eq(Tag.count)
-      expect(results.all? { |r| r.key?("id") }).to be_true
-      expect(results.all? { |r| r.key?("name") }).to be_true
-      expect(results.all? { |r| r.key?("categorization") }).to be_true
+      it "query_result" do
+        expect_query_result(:tags, :tag_count, :tag_count)
+        expect_result_resources_to_include_keys("resources", %w(id name categorization))
+      end
     end
   end
 
   context "Vm Tag subcollection" do
-    it "query all tags of a Vm with no tags" do
-      basic_authorize @cfme[:user], @cfme[:password]
+    context "query all tags of a Vm with no tags" do
+      before do
+        api_basic_authorize
 
-      @success = run_get @vm1_tags_url
+        run_get vm1_tags_url
+      end
 
-      expect(@success).to be_true
-      expect(@code).to eq(200)
-      expect(@result["name"]).to eq("tags")
-      expect(@result["subcount"]).to eq(0)
-      expect(@result["resources"].size).to eq(0)
+      it "empty_query_result" do
+        expect_empty_query_result(:tags)
+      end
     end
 
-    it "query all tags of a Vm" do
-      basic_authorize @cfme[:user], @cfme[:password]
+    context "query all tags of a Vm" do
+      before do
+        api_basic_authorize
 
-      @success = run_get @vm2_tags_url
+        run_get vm2_tags_url
+      end
 
-      expect(@success).to be_true
-      expect(@code).to eq(200)
-      expect(@result["name"]).to eq("tags")
-      expect(@result["count"]).to eq(Tag.count)
-      expect(@result["subcount"]).to eq(2)
-      expect(@result["resources"].size).to eq(2)
+      it "query_result" do
+        expect_query_result(:tags, 2, :tag_count)
+      end
     end
 
-    it "query all tags of a Vm and verify tag category and names" do
-      basic_authorize @cfme[:user], @cfme[:password]
+    context "query all tags of a Vm and verify tag category and names" do
+      def tag_paths
+        [tag1[:path], tag2[:path]]
+      end
+
+      before do
+        api_basic_authorize
 
-      @success = run_get "#{@vm2_tags_url}?expand=resources"
+        run_get "#{vm2_tags_url}?expand=resources"
```

Comment:
I notice that `run_get` is called in the `before`. Is this necessary?
Would you get rid of a few `context`s if this were part of the `it`?